Sources/DolbyIORTSUIKit/Private/Views/GridView/GridView.swift

```diff
@@ -0,0 +1,110 @@
+    }
+
+    private let viewModel: GridViewModel
+    private let layout: GridViewLayout
+    private let onVideoSelection: (StreamSource) -> Void
+    @State private var deviceOrientation: UIDeviceOrientation = UIDeviceOrientation.portrait
+
+    init(
+        viewModel: GridViewModel,
+        layout: GridViewLayout = .vertical(),
+        onVideoSelection: @escaping (StreamSource) -> Void
+    ) {
+        self.viewModel = viewModel
+        self.layout = layout
+        self.onVideoSelection = onVideoSelection
+    }
+
+    var body: some View {
+        GeometryReader { proxy in
+            VStack {
+                let screenSize = proxy.size
+                switch layout {
+                case .horizontal(rowsPortrait: let rowsPortrait, rowsLandscape: let rowsLandscape):
+                    gridHorizontal(availableHeight: screenSize.height, rowsCount: deviceOrientation.isPortrait ? rowsPortrait : rowsLandscape)
+                case .vertical(columnsPortrait: let columnsPortrait, columnsLandscape: let columnsLandscape):
+                    gridVertical(screenSize, deviceOrientation.isPortrait ? columnsPortrait : columnsLandscape)
+                }
+            }
+        }
+        .onRotate { newOrientation in
+            if !newOrientation.isFlat && newOrientation.isValidInterfaceOrientation {
+                deviceOrientation = newOrientation
+            }
+        }
+    }
+    
+    private func gridVertical(_ screenSize: CGSize, _ columnsCount: Int) -> some View {
+        let thumbnailSizeRatio = thumbnailRatioForColumnCount(columnCount: columnsCount)
+        let columns = [GridItem](repeating: GridItem(.flexible(), spacing: Layout.spacing1x), count: columnsCount)
+        return ScrollView {
+            LazyVGrid(columns: columns, alignment: .leading) {
+                ForEach(viewModel.allVideoViewModels, id: \.streamSource.id) { viewModel in
+                    let maxAllowedSubVideoWidth = screenSize.width * thumbnailSizeRatio
+                    let maxAllowedSubVideoHeight = screenSize.height * thumbnailSizeRatio
+                    
+                    VideoRendererView(
+                        viewModel: viewModel,
+                        maxWidth: maxAllowedSubVideoWidth,
+                        maxHeight: maxAllowedSubVideoHeight,
+                        contentMode: .aspectFit
+                    ) { source in
+                        onVideoSelection(source)
+                    }
+                    .frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .center)
+                }
+            }
+            .frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .center)
+        }
+    }
+
+    private func gridHorizontal(availableHeight: CGFloat, rowsCount: Int) -> some View {
```

Comment:
Minor style for matching gridVertical() method:
`gridHorizontal(_ availableHeight: CGFloat, _ rowsCount: Int)`